data-integration/main.bal

```diff
@@ -0,0 +1,103 @@
+            returns http:InternalServerError|http:Created|http:Conflict {
+        int[]|persist:Error result = self.dbClient->/employees.post([employee]);
+        if result is persist:Error {
+            if result is persist:AlreadyExistsError {
+                return http:CONFLICT;
+            }
+            return http:INTERNAL_SERVER_ERROR;
+        }
+        return http:CREATED;
+    }
+
+    isolated resource function post tasks(db:EmployeeTaskInsert task)
+            returns http:InternalServerError|http:Created|http:Conflict {
+        int[]|persist:Error result = self.dbClient->/employeetasks.post([task]);
+        if result is persist:Error {
+            if result is persist:AlreadyExistsError {
+                return http:CONFLICT;
+            }
+            return http:INTERNAL_SERVER_ERROR;
+        }
+        return http:CREATED;
+    }
+
+    isolated resource function get tasks(int taskId) returns db:EmployeeTask[]|error {
+        stream<db:EmployeeTask, persist:Error?> tasks = self.dbClient->/employeetasks;
+        return from db:EmployeeTask task in tasks
+            where task.taskId == taskId
+            select task;
+    }
+
+    isolated resource function get tasks/employee(int empId) returns db:EmployeeTask[]|error {
+        stream<db:EmployeeTask, persist:Error?> tasks = self.dbClient->/employeetasks;
+        return from db:EmployeeTask task in tasks
+            where task.employeeId == empId
+            select task;
+    }
+
+    isolated resource function get employees(int empId) returns db:Employee[]|error {
+        stream<db:Employee, persist:Error?> employees = self.dbClient->/employees;
+        return from db:Employee employee in employees
+            where employee.id == empId
+            select employee;
+    }
+
+    isolated resource function put tasks/[int taskId](db:EmployeeTaskUpdate emp) returns db:EmployeeTask|persist:Error {
+        return check self.dbClient->/employeetasks/[taskId].put(emp);
+    }
+
+    isolated resource function delete tasks/[int taskId]()
+            returns http:InternalServerError|http:NoContent|http:NotFound {
+        stream<db:EmployeeTask, persist:Error?> tasks = self.dbClient->/employeetasks;
+        db:EmployeeTask[]|persist:Error result = from db:EmployeeTask task in tasks
+            where task.taskId == taskId
+                && task.status == "COMPLETED"
+            select task;
+        if result is persist:Error {
+            if result is persist:NotFoundError {
+                return http:NOT_FOUND;
+            }
+            return http:INTERNAL_SERVER_ERROR;
+        }
+        foreach db:EmployeeTask task in result {
+            db:EmployeeTask|persist:Error deleteResult = self.dbClient->/employeetasks/[task.taskId].delete;
+            if deleteResult is persist:Error {
+                return http:INTERNAL_SERVER_ERROR;
+            }
+        }
+        return http:NO_CONTENT;
```

Comment:
No content should be return only if the stream has to `EmployeeTask` values. In here even we have `EmployeeTask` values we return `NO_CONTENT` response.